nilearn/_utils/tests/test_niimg_conversions.py

```diff
@@ -273,72 +273,68 @@ def test_check_niimg_wildcards():
     ):
         _utils.check_niimg(nofile_path_wildcards)
 
-    # First create some testing data
-    data_3d = np.zeros((40, 40, 40))
-    data_3d[20, 20, 20] = 1
-    img_3d = Nifti1Image(data_3d, np.eye(4))
 
-    data_4d = np.zeros((40, 40, 40, 3))
-    data_4d[20, 20, 20] = 1
-    img_4d = Nifti1Image(data_4d, np.eye(4))
+@pytest.mark.parametrize("shape", [(10, 10, 10), (10, 10, 10, 3)])
+@pytest.mark.parametrize(
+    "wildcards", [True, False]
+)  # (With globbing behavior or not)
+def test_check_niimg_wildcards(affine_eye, shape, wildcards):
+    # First create some testing data
+    img = Nifti1Image(np.zeros(shape), affine_eye)
 
-    #######
-    # Testing with an existing filename
-    with testing.write_tmp_imgs(img_3d, create_files=True) as filename:
-        assert_array_equal(
-            get_data(_utils.check_niimg(filename)), get_data(img_3d)
-        )
-    # No globbing behavior
-    with testing.write_tmp_imgs(img_3d, create_files=True) as filename:
+    with testing.write_tmp_imgs(img, create_files=True) as filename:
         assert_array_equal(
-            get_data(_utils.check_niimg(filename, wildcards=False)),
-            get_data(img_3d),
+            get_data(_utils.check_niimg(filename, wildcards=wildcards)),
+            get_data(img),
         )
 
-    #######
-    # Testing with an existing filename
-    with testing.write_tmp_imgs(img_4d, create_files=True) as filename:
-        assert_array_equal(
-            get_data(_utils.check_niimg(filename)), get_data(img_4d)
-        )
-    # No globbing behavior
-    with testing.write_tmp_imgs(img_4d, create_files=True) as filename:
-        assert_array_equal(
-            get_data(_utils.check_niimg(filename, wildcards=False)),
-            get_data(img_4d),
-        )
 
-    #######
+def test_check_niimg_wildcards_one_file_name(img_3d_zeros_eye):
+    tmp_dir = tempfile.tempdir + os.sep
```

Comment:
should we replace with tmp_path fixture?

Reply:
tried the most obvious ways and they did not work

thinking that maybe we could improve / fixturize 'testing.write_tmp_imgs', but that would be for another PR

Reply:
Ok agreed